mirrord/protocol/src/lib.rs

```diff
@@ -48,3 +49,41 @@ impl Deref for EnvVars {
         &self.0
     }
 }
+
+/// Used to identify if the remote pod is in a mesh context.
+///
+/// - Passed to the agent so we can handle the special case for selecting a network interface `lo`
+///   when we're mirroring with `istio`;
+/// - Used in the stealer iptables handling to add/detect special rules for meshes;
+///
+/// Can be converted to and from `String`, but the `from_str` just `panics` if you pass an invalid
+/// value (this use-case is hand-written, so it can't fail unless you add a new value and forget to
+/// handle it in the [`FromStr`] implementation).
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum MeshVendor {
+    Linkerd,
+    Istio,
+}
+
+impl fmt::Display for MeshVendor {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            MeshVendor::Linkerd => write!(f, "Linkerd"),
+            MeshVendor::Istio => write!(f, "Istio"),
+        }
+    }
+}
+
+impl FromStr for MeshVendor {
+    type Err = ();
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s {
+            "Linkerd" => Ok(MeshVendor::Linkerd),
+            "Istio" => Ok(MeshVendor::Istio),
+            invalid => panic!(
```

Comment:
I fear panic can cause problems with handling error of backwards compatibility if we add new mesh, so something like 
```rust
#[derive(Debug, Error)]
#[error("Invalid {0:?} option for MeshVendor")]
pub struct MeshVendorParseError(String);
```
and let clap print the error